src/components/views/SupplierLookup.tsx

```diff
@@ -0,0 +1,184 @@
+import '../../styles/supplierLookup.css';
+import { Textfield } from '../base/Textfield';
+import '../../styles/globalbtn.css';
+import { useState } from 'react';
+import { Supplier } from '../data/data';
+
+import SupplierTable from '../inputs/SupplierTable';
+
+interface SupplierLookupProps {
+  onClose: () => void;
+  onSupplierSelect: (supplier: Supplier) => void;
+}
+
+const SupplierLookup: React.FC<SupplierLookupProps> = ({
+  onClose,
+  onSupplierSelect,
+}) => {
+  const [name, setName] = useState('');
+  const [supplierIndex, setSIndex] = useState<number | null>(null);
+  const [city, setCity] = useState('');
+  const [filteredSuppliers, setFilteredSuppliers] = useState<Supplier[]>([]);
+  const [selectedSupplierIndex, setSelectedSupplierIndex] = useState<
+    number | null
+  >(null);
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
+  };
+
+  const handleIndexChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSIndex(event.target.value ? parseInt(event.target.value) : null);
+  };
+
+  const handleCityChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCity(event.target.value);
+  };
+
+  const handleSearch = (event: React.FormEvent) => {
+    event.preventDefault();
+    const hardcodedSuppliers: Supplier[] = [
+      { name: 'Andemis', supplierIndex: 1, city: 'San Francisco' },
+      { name: 'Anna', supplierIndex: 2, city: 'Machu Picchu' },
+      { name: 'Mathew', supplierIndex: 3, city: 'Kigali' },
+    ];
+
+    const filtered = hardcodedSuppliers.filter((supplier) => {
+      return (
+        (!name || supplier.name.toLowerCase().includes(name.toLowerCase())) &&
+        (supplierIndex === null || supplier.supplierIndex === supplierIndex) &&
+        (!city ||
+          (supplier.city &&
+            supplier.city.toLowerCase().includes(city.toLowerCase())))
+      );
+    });
+
+    setFilteredSuppliers(filtered);
+  };
+
+  const handleReset = () => {
+    setName('');
+    setSIndex(null);
+    setCity('');
+    setFilteredSuppliers([]);
+    setSelectedSupplierIndex(null);
+  };
+
+  const handleClose = () => {
+    if (onClose) onClose();
+  };
+
+  const handleSelectSupplier = (index: number) => {
+    setSelectedSupplierIndex(index);
+  };
+
+  const handleSubmit = () => {
+    if (selectedSupplierIndex !== null) {
+      const selectedSupplier = filteredSuppliers[selectedSupplierIndex];
```

Comment:
pls do not rely on index of array with this selection. Array can be easly reordered, no one guarante it will be allways in the same order. Eather rely on "supplierIndex" field which is unique or on the whole supplier object